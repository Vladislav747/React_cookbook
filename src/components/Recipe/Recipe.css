

.recipe {
  width: 200px;
  min-height: 250px;
  margin: 40px;
  background: #fff;
  padding: 15px;
  border-radius: 2px;
  flex-basis: calc(33% - 16px);
  -webkit-flex-basis: calc(33% - 16px);
  margin: 8px;
  overflow-wrap: break-word;
  box-shadow: 0 0 15px 1px rgba(0, 0, 0, 0.4);
}

.recipe-name{
  text-align: center;
  padding: 16px 0px;
}

.recipe-description{
  text-align: center;
  padding-bottom: 16px;
}

.recipe-image {
  overflow: hidden;
  border-radius: 2px 2px 0 0;
  max-height: 200px;
}

.recipe-image img {
  cursor: zoom-in;
  width: 100%;
  height: auto;
  transition: transform .3s ease-in;
  transform: scale(1);
}

.recipe-image img:hover {
  transform: scale(1.1);
}

.recipe-action{
  justify-content: center;
  display: flex;
  padding:16px;
}

.like-block{
  margin: 20px 0 !important;
}