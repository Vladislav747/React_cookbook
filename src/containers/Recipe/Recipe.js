import { connect } from "react-redux";
import { bindActionCreators } from "redux";
// import * as  from 'redux/actions/cart';
import { Recipe } from "components";

const mapStateToProps = ({ cart }, { id }) => ({
    addedCount: cart.items.reduce(
        (count, book) => count + (book.id === id ? 1 : 0),
        0
    ),
});

const mapDispatchToProps = (dispatch) => ({
    ...bindActionCreators({}, dispatch),
});

export default connect(mapStateToProps, mapDispatchToProps)(Recipe);
