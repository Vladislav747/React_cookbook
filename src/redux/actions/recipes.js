export const setRecipes = recipes => ({
  type: 'SET_BOOKS',
  payload: recipes
});
