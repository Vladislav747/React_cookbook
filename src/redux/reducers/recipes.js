const initialState = {
    isReady: false,
    items: null,
};

export default (state = initialState, action) => {
    switch (action.type) {
        case "SET_RECIPES":
            return {
                ...state,
                items: action.payload,
                isReady: true,
            };
        case "SET_IS_READY":
            return {
                ...state,
                isReady: action.payload,
            };
        case "DELETE_RECIPE":
            return {
                ...state,
                items: state.items.filter((o) => o.id !== action.payload),
            };
        //TODO доделаnь
        case "ADD_LIKE":
            return {
                ...state,
                items: state.items.map((reduxItem) => {
                    if (reduxItem.id === action.payload) {
                        reduxItem.likes = reduxItem.likes + 1;
                    }
                }),
            };
        default:
            return state;
    }
};
