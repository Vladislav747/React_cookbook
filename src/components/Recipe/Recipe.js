import React from "react";
import { PropTypes } from "prop-types";
import { Card, Button, Image, Icon, Label } from "semantic-ui-react";

import "./Recipe.css";

const Recipe = ({
    deleteRecipe,
    addLike,
    ingredients,
    title,
    author,
    image,
    likes,
    id,
}) => (
    <Card>
        <Image src={image} wrapped ui={false} alt="Фото рецепта" />
        <Card.Content>
            <Card.Header>
                <h2>{title}</h2>
            </Card.Header>
            <Card.Meta>
                <span className="date">{author}</span>
            </Card.Meta>
            <Card.Description>
                <h5>Ингредиенты</h5>
                <ul>
                    {ingredients.map((el, index) => (
                        <li key={index}>{el}</li>
                    ))}
                </ul>
            </Card.Description>
        </Card.Content>
        <Card.Content extra>
            <Button
                className="btnDeleteRecipe"
                onClick={deleteRecipe.bind(this, id)}
                color="red"
            >
                УДАЛИТЬ РЕЦЕПТ
            </Button>
            <div className="like-block" labelPosition="right">
                <Button color="orange" onClick={addLike.bind(this, id)}>
                    <Icon name="heart" />
                    Like
                </Button>
                <Label as="a" basic color="orange" pointing="left">
                    {likes}
                </Label>
            </div>
        </Card.Content>
    </Card>
);

Recipe.propTypes = {
    ingredients: PropTypes.array,
    title: (props, propName) =>
        typeof props[propName] !== "string" && props[propName].length > 50
            ? new Error("A title must be a string and over 50 characters")
            : null,
};

export default Recipe;
