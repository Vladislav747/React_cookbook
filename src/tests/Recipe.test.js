// Recipe.test.js
import React from "react";
import { create, update } from "react-test-renderer";

import recipes from "../../public/recipesArray.js";
import { addLike, deleteRecipe } from "../redux/actions/recipes.js";
import Recipe from "../components/Recipe/Recipe";

it("Component Recipe renders without crashing", () => {
    let tree = create(
        <Recipe
            ingredients={recipes[0].ingredients}
            title={recipes[0].title}
            author={recipes[0].author}
            image={recipes[0].image}
            addLike={addLike}
            deleteRecipe={deleteRecipe}
        />
    );

    expect(tree.toJSON()).toMatchSnapshot();
});
