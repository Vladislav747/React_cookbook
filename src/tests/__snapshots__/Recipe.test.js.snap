// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="ui card"
  href={undefined}
  onClick={[Function]}
>
  <div
    className="image"
    href={undefined}
  >
    <img
      alt="Фото рецепта"
      src="https://yummybook.ru/upload/images/recipe/item/9738/image_large_39865f3c29eab214097779178_97779178.jpg"
    />
  </div>
  <div
    className="content"
  >
    <div
      className="header"
    >
      <h2>
        Детское яблочное пюре
      </h2>
    </div>
    <div
      className="meta"
    >
      <span
        className="date"
      >
        Иванова Ольга Сергеевна
      </span>
    </div>
    <div
      className="description"
    >
      <h5>
        Ингредиенты
      </h5>
      <ul>
        <li>
          Яблоко 2-3 шт
        </li>
        <li>
          Вода 250 мл
        </li>
      </ul>
    </div>
  </div>
  <div
    className="extra content"
  >
    <button
      className="ui red button btnDeleteRecipe"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      УДАЛИТЬ РЕЦЕПТ
    </button>
    <div
      className="ui right labeled button like-block"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={0}
    >
      <button
        className="ui orange button"
        disabled={undefined}
        onClick={[Function]}
        role="button"
        tabIndex={undefined}
      >
        <i
          aria-hidden="true"
          className="heart icon"
        />
        Like
      </button>
      <a
        className="ui orange left pointing basic label"
        onClick={[Function]}
      />
    </div>
  </div>
</div>
`;
